spring-pulsar-spring-boot-autoconfigure/src/main/java/org/springframework/pulsar/autoconfigure/PulsarProperties.java

```diff
@@ -1323,6 +1323,21 @@ public static class Listener {
 		 */
 		private SchemaType schemaType;
 
+		/**
+		 * Max number of messages for batch.
+		 */
+		private int maxNumMessages = -1;
+
+		/**
+		 * Max number of bytes for batch.
+		 */
+		private int maxNumBytes = 10 * 1024 * 1024;
+
+		/**
+		 * Timeout for batch.
+		 */
+		private int batchTimeout = 100;
```

Comment:
I think we should move this to a `Duration` because it is ambiguous as to what it is currently (millis, seconds?).

Reply:
https://github.com/spring-projects-experimental/spring-pulsar/issues/119 for later